src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -86,7 +86,15 @@ private function doProcessValue($value, $isRoot = false)
             if ($ref = $this->getAutowiredReference($value)) {
                 return $ref;
             }
-            $this->container->log($this, $this->createTypeNotFoundMessage($value, 'it'));
+            $message = $this->createTypeNotFoundMessage($value, 'it');
+
+            if (ContainerBuilder::RUNTIME_EXCEPTION_ON_INVALID_REFERENCE === $value->getInvalidBehavior()) {
+                $this->container->register((string) $value, $value->getType())
```

Comment:
a next service referencing the same value in a stricter way would now use this service as it exists. This looks wrong to me.

Reply:
and even if it also use RUNTIME_EXCEPTION_ON_INVALID_REFERENCE and so would be fine with a throwing definition, it would still need a different error message for the other service, and so a *different* throwing definition.

Reply:
The behavior is on the reference, not on the definition. `DefinitionErrorExceptionPass` will throw if any references exist that are not `RUNTIME_EXCEPTION_ON_INVALID_REFERENCE`.
Or I missed your point.

Reply:
ok, fixed

Reply:
please add tests covering this, to ensure that we keep getting warned with the right error message in the future, as this is a tricky case.

Reply:
I added a comment instead.